lib/debug/repl.cpp

```diff
@@ -1,1028 +1,217 @@
 #include "repl.hpp"
-
-#include <fstream>
-#include <iostream>
-#include <sstream>
-#include <thread>
-
-#include <rapidjson/document.h>
-#include <rapidjson/error/en.h>
-#include <rapidjson/istreamwrapper.h>
+#include "repl_logger.hpp"
+
+#include "commands/adc_commands.hpp"
+#include "commands/can_commands.hpp"
+#include "commands/gpio_commands.hpp"
+#include "commands/i2c_commands.hpp"
+#include "commands/pwm_commands.hpp"
+#include "commands/spi_commands.hpp"
+#include "commands/uart_commands.hpp"
+#include <core/wall_clock.hpp>
+#include <io/hardware_adc.hpp>
+#include <io/hardware_can.hpp>
+#include <io/hardware_i2c.hpp>
 
 namespace hyped::debug {
 
-Repl::Repl(core::ILogger &logger)
-    : logger_(logger),
-      i2c_(),
-      spi_(),
-      pwm_(),
-      adc_(),
-      uart_(),
-      gpio_(io::HardwareGpio(logger))
-{
-}
-
-void Repl::run()
-{
-  while (true) {
-    handleCommand();
-  }
-}
-
-std::optional<std::unique_ptr<Repl>> Repl::fromFile(const std::string &path)
+std::optional<std::shared_ptr<Repl>> Repl::create(core::ILogger &logger,
+                                                  Terminal &terminal,
+                                                  const std::string &filename)
```

Comment:
idk if its worth mentioning somewhere exactly what the input file is, but this might be better as a wiki thing